### PR TITLE
🌝 Bounty page components

### LINKED ISSUE #1941: Bounty Details Page
## Scope
- [x] #1956
- [x] #1957
- [x] #1958

## Designs
[Link to Figma](https://www.figma.com/file/jZwIoaxOkTxNFMJNzssBgv/Bounty?node-id=248%3A120650)

### LINKED ISSUE #1956: Bounty page components
- [x] #1972
- [x] #1973
- [x] #1974
- [x] #1975
- [x] #1976
- [x] #1959

### PR DESCRIPTION
**Pull request by RafalPloszka**
Closes #1956 
Closes #1941

### CODE REVIEW

**Comment by RafalPloszka** [comment]
:warning:  Important note for code review
I made some rearrangements in files/directories structure, I describe them here for easier code review:

1.  New directory `bounty/components/BountyMainComponents`.
Components with the main part of bounty page according to its stage. `BountyExpired` and `BountyFailed` were moved there and I added corresponding components for other stages.

2.  New directory `bounty/components/tabsSets`:
`ResultsTabs` for bounties in Failed and Succesful stages, `CommonTabs` for the rest.
(`ExpiredTabs` was deleted and now `CommonTabs` is used instead. `FailedTabs` was deleted and now `ResultsTabs` is used instead.)

3.  New directory `bounty/components/BountyTiles`:
`FundingTiles` for Funding stage, `ResultsTiles` for Failed/Successful stage, `CommonTiles` for the rest.
(`ExpiredTiles` was deleted and now `CommonTiles` is used instead. `FailedTiles` was deleted and now `ResultsTiles` is used instead.)

4. New directory `bounty/components/modalsButtons`:
Buttons for calling buttons. `AddBountyModalButton`, `SubmitWorkButton` and 'WithdrawStakeButton` were moved here and I added buttons for calling the rest of bounties modals

File: packages/ui/src/bounty/components/BountyMainComponents/BountyFunding.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <>
+      <MainPanel>
+        <FundingTiles bounty={bounty} />
+        <ContentWithSidePanel>
+          <BountyTab bounty={bounty} />
+          <RowGapBlock gap={4}>
+            <BountySidebar
+              contributors={bounty.contributors}
+              stage="funding"
+              periodsLengths={{
```
Comment (Qcza) [inline packages/ui/src/bounty/components/BountyMainComponents/BountyFunding.tsx:27]: Move this object to `useMemo`.

File: packages/ui/src/bounty/components/BountyMainComponents/BountyJudgement.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <ContentWithSidePanel>
+          {active === 'Bounty' && <BountyTab bounty={bounty} />}
+          {active === 'Works' && <WorkTab bountyId={bounty.id} />}
+          <RowGapBlock gap={4}>
+            <BountySidebar
+              contributors={bounty.contributors}
+              entrants={entrants}
+              withdrawals={withdrawns}
+              stage="judgement"
+              periodsLengths={{
```
Comment (Qcza) [inline packages/ui/src/bounty/components/BountyMainComponents/BountyJudgement.tsx:38]: Move this object to `useMemo`

File: packages/ui/src/bounty/components/BountyMainComponents/BountyWorking.tsx [hunk 3]
```diff
@@ -1,39 +1,42 @@
...
-          {active === 'Winners' && <WinnersTab bounty={bounty} />}
           <RowGapBlock gap={4}>
             <BountySidebar
               contributors={bounty.contributors}
-              hidePeriods
-              stage="withdrawal"
+              entrants={entrants}
+              withdrawals={withdrawns}
+              stage="working"
               periodsLengths={{
```
Comment (Qcza) [inline packages/ui/src/bounty/components/BountyMainComponents/BountyWorking.tsx:38]: Move it to `useMemo`

File: packages/ui/src/bounty/components/BountyPreviewHeader/BountyPreviewHeader.tsx [hunk 4]
```diff
@@ -1,10 +1,18 @@
+import BN from 'bn.js'
```
Comment (Qcza) [inline packages/ui/src/bounty/components/BountyPreviewHeader/BountyPreviewHeader.tsx:1]: Apply `React.memo` to all components in this file.

File: packages/ui/src/bounty/components/BountyMainComponents/BountyExpired.tsx [hunk 5]
```diff
@@ -46,9 +41,10 @@ export const BountyExpired = ({ bounty }: Props) => {
             <BountySidebar
               entrants={entrants}
               contributors={bounty.contributors}
+              withdrawals={withdrawns}
               stage="expired"
               periodsLengths={{
```
Comment (Qcza) [inline packages/ui/src/bounty/components/BountyMainComponents/BountyExpired.tsx:46]: Move it to `useMemo`

File: packages/ui/src/bounty/components/BountyPreviewHeader/BountyPreviewHeader.tsx [hunk 6]
```diff
@@ -140,20 +159,24 @@ const SuccessfulStageButtons = ({ bounty, activeMember, t }: BountyHeaderButtons
       <ButtonGhost size="large">
         <BellIcon /> {t('common:buttons.notifyAboutChanges')}
       </ButtonGhost>
-      {winner && <ButtonGhost size="large">{t('common:buttons.claimReward')}</ButtonGhost>}
-      {(passed || isContributor) && <ButtonGhost size="large">{t('common:buttons.withdrawStake')}</ButtonGhost>}
+      {winnerConditions && <ClaimRewardButton bountyId={bounty.id} entryId={entryId} reward={new BN(reward)} />}
```
Comment (Qcza) [inline packages/ui/src/bounty/components/BountyPreviewHeader/BountyPreviewHeader.tsx:162]: Do not call the constructor in the component prop if the component you pass this prop to uses `React.memo`.

**Review by Qcza — CHANGES_REQUESTED** [review]
Please apply `React.memo` to components that receive props.

**Review by Qcza — APPROVED** [review]

**Pull request merged**